src/contacts/EditContact/EditContactContainer.js

```diff
@@ -1,29 +1,34 @@
 import React, { Component } from 'react';
 import { FormattedMessage } from 'react-intl';
 import PropTypes from 'prop-types';
-
+import ReactRouterPropTypes from 'react-router-prop-types';
 import { get } from 'lodash';
 
+import { stripesConnect } from '@folio/stripes/core';
+
 import {
   categoriesResource,
   contactResource,
 } from '../../common/resources';
 import { saveContact } from './util';
 import EditContact from './EditContact';
 import { getBackQuery } from '../../common/utils/createItem';
+import { DICT_CATEGORIES } from '../../common/constants';
 
 class EditContactContainer extends Component {
   static manifest = Object.freeze({
     contact: contactResource,
-    categories: categoriesResource,
+    [DICT_CATEGORIES]: categoriesResource,
     query: {},
   });
 
   onClose = (contactId = this.props.match.params.id) => {
-    const { orgId, mutator } = this.props;
+    const { orgId, mutator, baseUrl, history } = this.props;
     const query = getBackQuery(orgId, contactId, 'contacts');
 
-    mutator.query.replace(query);
+    if (baseUrl) {
+      history.push(`${baseUrl}/${contactId}/view`);
```

Comment:
actually, that the case what I've asked to avoid. It's hard to say why if `baseUrl` exists we need to go there and to do another thing in other cases. I think the same `onClose` prop should be passed, as in view ctrl

Reply:
fixed